### PR TITLE
Support qlora in CPU

### PR DESCRIPTION
**Pull request by yangw1234**
## Description

Support qlora in CPU

### CODE REVIEW

**Comment by yangw1234** [comment]
@hzjane please refer this PR

File: python/llm/example/GPU/QLoRA-FineTuning/qlora_finetuning.py [hunk 1]
```diff
@@ -43,13 +42,16 @@
...
-                                                modules_to_not_convert=["lm_head"],)
-    model = model.to('xpu')
-    model.gradient_checkpointing_enable()
-    model = prepare_model_for_kbit_training(model)
+                                                )
+    model = model.to("cpu")
+    # model.gradient_checkpointing_enable()
+    model = prepare_model_for_kbit_training(model, use_gradient_checkpointing=False)
+
+    model.enable_input_require_grads()
```
Comment (yangw1234) [inline python/llm/example/GPU/QLoRA-FineTuning/qlora_finetuning.py:54]: Somehow both setting `use_gradient_checkpoint=False` and calling `model.enable_input_require_grads()` is necessary. I still have not figure out why.

File: python/llm/example/GPU/QLoRA-FineTuning/qlora_finetuning.py [hunk 2]
```diff
@@ -71,8 +73,8 @@
             max_steps=200,
             learning_rate=2e-4,
             save_steps=100,
-            fp16=True,
-            logging_steps=20,
+            bf16=True,
```
Comment (yangw1234) [inline python/llm/example/GPU/QLoRA-FineTuning/qlora_finetuning.py:76]: must use "bf16" in for cpu

**Review by jason-dai — APPROVED** [review]
LGTM

**Comment by yangw1234** [comment]
ARC test is unstable. Merge it for now.

**Pull request merged**